test/modern_bpf/test_suites/syscall_exit_suite/creat_x.cpp

```diff
@@ -0,0 +1,116 @@
+#include "../../event_class/event_class.h"
+
+#ifdef __NR_creat
+
+#if defined(__NR_fstat) && defined(__NR_unlinkat) && defined(__NR_close)
+TEST(SyscallExit, creatX_success)
+{
+	auto evt_test = new event_test(__NR_creat, EXIT_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	const char* path = "tmp_file";
+	mode_t mode = S_IRWXU;
+	int fd = syscall(__NR_creat, path, mode);
+	assert_syscall_state(SYSCALL_SUCCESS, "creat", fd, NOT_EQUAL, -1);
+
+	/* Call `fstat` to retrieve the `dev` and `ino`. */
+	struct stat file_stat;
+	assert_syscall_state(SYSCALL_SUCCESS, "fstat", syscall(__NR_fstat, fd, &file_stat), NOT_EQUAL, -1);
```

Comment:
This is an extreme nitpick, but do we really need to use `syscall` to retrieve the dev and inode numbers? Can't we just `stat`? Don't get me wrong, it's fine as is and obviously calling syscalls directly is what we need for the tests, it's just a little weird seeing syscalls also being used when you just want to get some value from the filesystem and you could use an existing wrapper 🤷🏻‍♂️

Reply:
I guess this could also be asked of calls to `close` and `unlink`, it's nothing major, I'm just annoying as usual.

Reply:
That's a good point, in these tests I usually prefer to use more low levels calls as possible since we are working on different architectures and I'm not completely sure about the call behind the wrapper in that specific architecture, and as a personal taste usually in these cases I prefer to not use the glibc wrappers but the bare `syscall`, again not a strong opinion on that, maybe we can do all this stuff as a final cleanup, if we want, since this involves many different PR and not only this one